### PR TITLE
x.py: support source packaging

### PR DESCRIPTION
**Pull request by tisonkun**
Try out with: `./x.py package source -v 2.1.0` in a staging branch.

### CODE REVIEW

File: x.py [hunk 1]
```diff
@@ -131,6 +131,36 @@ def cppcheck(args):
...
+    version = args.release_version.strip()
+    if SEMVER_REGEX.match(version) is None:
+        raise RuntimeError(f"Kvrocks version should follow semver spec, got: {version}")
+    
+    # 0. Write input version to VERSION file
+    with open('VERSION', 'w+') as f:
+        f.write(version)
+
+    # 1. Git commit and tag
+    git = find_command('git', msg='git is required for source packaging')
```
Comment (tisonkun) [inline x.py:144]: We don't push anything to remote here. The release manager should manually `git push --tags` to push the tag.

In my opinion the workflow should be:

1. Checkout a staging branch - if we do lightweight release, it's possible to run on `unstable` branch.
2. Run `./x.py package source` to get the source tarball, `.asc` sign file, `.sha512` checksum file.
3. Copy those files to SVN (https://dist.apache.org/repos/dist/dev/incubator/kvrocks).
4. Manually push tags (maybe we can have a `-rcN` tailer, but I don't make it too complex now.
5. Calling a vote.

File: x.py [hunk 2]
```diff
@@ -131,6 +131,36 @@ def cppcheck(args):
...
+    run([git, 'tag', '-a', f'v{version}', '-m', '[source-release] copy for tag v{version}'], stdout=subprocess.PIPE)
+
+    tarball = f'apache-kvrocks-{version}.tar.gz'
+    # 2. Create the source tarball
+    output = run([git, 'ls-files'], stdout=subprocess.PIPE)
+    run(['xargs', 'tar', '-czf', tarball], stdin=output, stdout=subprocess.PIPE)
+
+    # 3. GPG Sign
+    gpg = find_command('gpg', msg='gpg is required for source packaging')
+    run([gpg, '--detach-sign', '--armor', tarball], stdout=subprocess.PIPE)
```
Comment (tisonkun) [inline x.py:155]: FYI - You can verify signature by `gpg --verify ${tarball}.asc ${tarball}`.

File: x.py [hunk 3]
```diff
@@ -131,6 +131,36 @@ def cppcheck(args):
...
+
+    # 4. Generate sha512 checksum
+    sha512sum = find_command('sha512sum', msg='sha512sum is required for source packaging')
+    output = run([sha512sum, tarball], stdout=subprocess.PIPE)
+    payload = output.read().decode().strip()
+    with open(f'{tarball}.sha512', 'w+') as f:
+        f.write(payload)
```
Comment (tisonkun) [inline x.py:162]: FYI - You can verify checksum by `sha512sum -c ${tarball}.sha512`.

File: x.py [hunk 4]
```diff
@@ -131,6 +131,36 @@ def cppcheck(args):
...
     run([command, *options, *sources])
 
+def package_source(args):
+    version = args.release_version.strip()
+    if SEMVER_REGEX.match(version) is None:
+        raise RuntimeError(f"Kvrocks version should follow semver spec, got: {version}")
+    
+    # 0. Write input version to VERSION file
+    with open('VERSION', 'w+') as f:
+        f.write(version)
```
Comment (PragmaTwice) [inline x.py:141]: I think we can just discard this file and pass the version to cmake directly, how do you think about this?

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks @tisonkun

**Review by PragmaTwice — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your reviews.

Merging...

**Pull request merged**